Production/plugins/TauTupleProducer.cc

```diff
@@ -1068,23 +1030,52 @@ class TauTupleProducer : public edm::EDAnalyzer {
-	   tauTuple().sv_x.push_back(static_cast<float>(SV->position().x()));
-           tauTuple().sv_y.push_back(static_cast<float>(SV->position().y()));
-           tauTuple().sv_z.push_back(static_cast<float>(SV->position().z()));
-           tauTuple().sv_t.push_back(static_cast<float>(SV->t()));
-	   tauTuple().sv_xE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(0,0))));
-           tauTuple().sv_yE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(1,1))));
-           tauTuple().sv_zE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(2,2))));
-           tauTuple().sv_tE.push_back(static_cast<float>(SV->tError()));
-	   tauTuple().sv_chi2.push_back(static_cast<float>(SV->vertexChi2()));
-           tauTuple().sv_ndof.push_back(static_cast<float>(SV->vertexNdof()));
+
+        auto findMatch = [&](const edm::Ptr<reco::Candidate>& recocand) { 
+            for(size_t cand_idx = 0; cand_idx < cands.size(); ++cand_idx) {
+                auto pfcand = dynamic_cast<const reco::Candidate*>(cands.at(cand_idx).candidate);
+                if(pfcand == &(*recocand))
+                   return static_cast<int>(cand_idx);
+            }
+            return -1;
+        };
+
+   	auto collectCands = [&](const reco::VertexCompositePtrCandidate& sv) {
+            std::set<int> cand_indices;
+            for(size_t it = 0; it < sv.numberOfSourceCandidatePtrs(); it++) {
+                const edm::Ptr<reco::Candidate>& recocand = sv.sourceCandidatePtr(it);
+                int match_idx = findMatch(recocand);
+                if(match_idx >= 0)
+                   cand_indices.insert(match_idx);
+            }
+            return cand_indices;
+        };
+
+	int nsv = 0;
+        for(const auto& sv_ptr : sv_col) {
+            const reco::VertexCompositePtrCandidate* SV = sv_ptr.obj;
+            auto candsIdx = collectCands(*SV);
+            if(candsIdx.size() > 0){
+	        tauTuple().sv_x.push_back(static_cast<float>(SV->position().x()));
+                tauTuple().sv_y.push_back(static_cast<float>(SV->position().y()));
+           	tauTuple().sv_z.push_back(static_cast<float>(SV->position().z()));
+           	tauTuple().sv_t.push_back(static_cast<float>(SV->t()));
+	   	tauTuple().sv_xE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(0,0))));
+           	tauTuple().sv_yE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(1,1))));
+           	tauTuple().sv_zE.push_back(static_cast<float>(std::sqrt(SV->vertexCovariance(2,2))));
+           	tauTuple().sv_tE.push_back(static_cast<float>(SV->tError()));
+	   	tauTuple().sv_chi2.push_back(static_cast<float>(SV->vertexChi2()));
+           	tauTuple().sv_ndof.push_back(static_cast<float>(SV->vertexNdof()));
+		for(int cand_idx : candsIdx) {
+                    tauTuple().sv_cands_svIdx.push_back(sv_ptr.index);
+                    tauTuple().sv_cands_pfcandIdx.push_back(cand_idx);
+  		}
+	        ++nsv;
+            }
```

Comment:
@ofivite correct me if I am wrong, in this way -> SVs that matched to PFCand that selected within cone will be filled only, so then the difference I was thinking to also add the guys that are within the cone but do not have any pfcands. Also, this structure with indexes is a bit weird, I would have indexes for Pfcands pointing SV then it is just <int>

Reply:
we have chosen SV<->pfCand mapping as a separate vectors, because we were not sure if pfCand->SV is guaranteed to be unique. Do you know if it is unique by design?